code/modules/mob/mob_helpers.dm

```diff
@@ -224,12 +227,31 @@
 	var/max = strength*world.icon_size
 	var/min = -(strength*world.icon_size)
 
-	for(var/i in 0 to duration-1)
-		if (i == 0)
-			animate(C, pixel_x=rand(min,max), pixel_y=rand(min,max), time=1)
+	//How much time to allot for each pixel moved
+	var/time_scalar = (1 / world.icon_size) * TILES_PER_SECOND
+	var/last_x = oldx
+	var/last_y = oldy
+
+	var/time_spent = 0
+	while(time_spent < duration)
+		//Get a random pos in our box
+		var/x_pos = rand(min, max) + oldx
+		var/y_pos = rand(min, max) + oldy
+
+		//We take the smaller of our two distances so things still have the propencity to feel somewhat jerky
+		var/time = round(max(min(abs(last_x - x_pos), abs(last_y - y_pos)) * time_scalar, 1))
```

Comment:
are you sure you dont want clamp? this is hard to read

Reply:
I remember someone mentioning fucked behavior with clamp in the vein of like, the smallest arg is taken as the min or something. I can look deeper into it if you'd like

Reply:
weird, i'd've just expected it to error

Reply:
scratch that, clamp would be ok here

Reply:
oh actually no it'd make applying the scalar a headache, leaving it as is if that's ok